Sources/SwiftInspectorVisitors/GenericRequirementVisitor.swift

```diff
@@ -0,0 +1,82 @@
+// Created by Dan Federman on 1/26/21.
+//
+// Copyright © 2021 Dan Federman
+//
+// Distributed under the MIT License
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+import Foundation
+import SwiftSyntax
+
+public final class GenericRequirementVisitor: SyntaxVisitor {
+
+  public var genericRequirements = [GenericRequirement]()
+
+  public override func visit(_ node: SameTypeRequirementSyntax) -> SyntaxVisitorContinueKind {
+    genericRequirements.append(GenericRequirement(from: node))
+    return .visitChildren
+  }
+
+  public override func visit(_ node: ConformanceRequirementSyntax) -> SyntaxVisitorContinueKind {
+    genericRequirements.append(GenericRequirement(from: node))
+    return .visitChildren
+  }
+
+}
+
+public struct GenericRequirement: Codable, Equatable {
```

Comment:
This type doesn't _need_ to be Codable, but I figure it might be a good thing to make these types capable of being persisted easily.